lib/utils/formulaUtils.jsx

```diff
@@ -0,0 +1,40 @@
+import { formatChemicalFormula } from "./formatting";
+
+export function countNumberOfAtoms(formula) {
+  // split on capital letters to get element+number strings
+  var elnum = formula.split(/(?=[A-Z])/);
+  var num = 0;
+  elnum.forEach((v) => {
+    let match = v.match(/\d+/);
+    let n = match == null ? 1 : parseInt(match[0]);
+    num += n;
+  });
+  return num;
+}
+
+export function countNumberOfElements(formula) {
+  return formula.split(/(?=[A-Z])/).length;
+}
+
+export function formatTitle(formulaStr, id, method) {
```

Comment:
This one, I think, is not very generally usable, and quite specific to the MC3D frontend. I would keep it there.